mvx-esdt-safe/tests/mvx_esdt_safe_blackbox_unit_tests.rs

```diff
@@ -723,3 +897,125 @@ fn execute_operation_success() {
                 .is_empty());
         })
 }
+
+#[test]
+fn execute_operation_burn_mechanism_without_deposit() {
+    let mut state = MvxEsdtSafeTestState::new();
+    state.deploy_contract_with_roles();
+
+    let token_data = EsdtTokenData {
+        amount: BigUint::from(100u64),
+        ..Default::default()
+    };
+
+    let payment =
+        OperationEsdtPayment::new(TokenIdentifier::from(TRUSTED_TOKEN_IDS[0]), 0, token_data);
+
+    let operation_data = OperationData::new(1, OWNER_ADDRESS.to_managed_address(), None);
+
+    let operation = Operation::new(
+        TESTING_SC_ADDRESS.to_managed_address(),
+        vec![payment].into(),
+        operation_data,
+    );
+
+    let operation_hash = state.get_operation_hash(&operation);
+    let hash_of_hashes = ManagedBuffer::new_from_bytes(&sha256(&operation_hash.to_vec()));
+
+    state.deploy_header_verifier();
+    state.deploy_testing_sc();
+    state.set_esdt_safe_address_in_header_verifier(ESDT_SAFE_ADDRESS);
+
+    let operations_hashes = MultiValueEncoded::from(ManagedVec::from(vec![operation_hash.clone()]));
+
+    state.deploy_chain_config(SovereignConfig::default_config());
+    state.register_operation(ManagedBuffer::new(), &hash_of_hashes, operations_hashes);
+
+    state
+        .world
+        .query()
+        .to(HEADER_VERIFIER_ADDRESS)
+        .whitebox(header_verifier::contract_obj, |sc| {
+            let operation_hash_whitebox = ManagedBuffer::new_from_bytes(&operation_hash.to_vec());
+            let hash_of_hashes =
+                ManagedBuffer::new_from_bytes(&sha256(&operation_hash_whitebox.to_vec()));
+
+            assert!(
+                sc.operation_hash_status(&hash_of_hashes, &operation_hash_whitebox)
+                    .get()
+                    == OperationHashStatus::NotLocked
+            );
+        });
+
+    state.set_token_burn_mechanism(TRUSTED_TOKEN_IDS[0], None);
+
+    state.execute_operation(hash_of_hashes, operation.clone(), None);
```

Comment:
This `execute_operation` is for that special error case when you try to mint more than was burned. Error is `None` good. But you have to check balances and also hash is header verifier that was removed.